### PR TITLE
FallingBlock auto expire setting

### PR DESCRIPTION
**Pull request by Owen1212055**
This gives the ability to disable falling blocks from being able to expire automatically.
This also saves with nbt under ``DisableAutoExpire``, although I am not sure this behavior is needed.

### CODE REVIEW

**Comment by imDaniX** [comment]
Maybe a couple of options to actually set the expire timers instead? Also, what's the use-case for this nbt?

**Comment by Owen1212055** [comment]
> Maybe a couple of options to actually set the expire timers instead? Also, what's the use-case for this nbt?

What do you propose? So like modifying it without changing the entity age? (Currently set age modifies this timer and the entity age) And I’m not sure, since I don’t know if new fields like this should be serialized into the entity or not.

**Comment by imDaniX** [comment]
> What do you propose? So like modifying it without changing the entity age? (Currently set age modifies this timer and the entity age) And I’m not sure, since I don’t know if new fields like this should be serialized into the entity or not.

Sorry, actually did read an entire PR wrong.

**Comment by Owen1212055** [comment]
~~Simplified by removing the tag, this is now a simple boolean.
Current variables for falling blocks by spigot aren't retained either... so I felt perhaps it wasn't needed to keep it.~~

This is a rather simple API addition and is ready.

**Comment by Owen1212055** [comment]
Rebased

File: patches/server/0877-FallingBlock-auto-expire-setting.patch [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ 
+     }
+ 
+@@ -360,6 +362,10 @@ public class FallingBlockEntity extends Entity {
+             int srcZ = nbt.getInt("SourceLoc_z");
+             this.setOrigin(new org.bukkit.Location(level.getWorld(), srcX, srcY, srcZ));
+         }
++
++        if (nbt.contains("Paper.AutoExpire")) {
++            this.autoExpire = nbt.contains("Paper.AutoExpire");
```
Comment (kennytv) [inline patches/server/0877-FallingBlock-auto-expire-setting.patch]: get the boolean value

**Review by kennytv — CHANGES_REQUESTED** [review]

File: patches/server/0877-FallingBlock-auto-expire-setting.patch (hunk 1 above)
Comment (Owen1212055) [reply patches/server/0877-FallingBlock-auto-expire-setting.patch]: LOL nice, will do

**Comment by Owen1212055** [comment]
Rebased and fixed

**Review by kennytv — APPROVED** [review]

**Pull request merged**